### ISSUE TITLE
Revise not executing identical lines of code

### ISSUE THREAD

**Issue by dcabecinhas** [timholy/Revise.jl#511]
`Revise.jl` seems to be silently not executing identical instructions even though they have side-effects.

_Includeting_ a file with contents
```julia
a=1
a += a
println("a:", a)
a += a
println("a:", a)
a += a
println("a:", a)
println("a final:", a)
```
results in 
```
julia> includet("test_revise.jl")
a:2
a final:2
```
which is not what is expected.

The same situation happens when editing a module, although not when first loading it.

System info: (also occurs with Julia 1.4.2)
```
Julia Version 1.5.0-rc1.0
Commit 24f033c951 (2020-06-26 20:13 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin19.4.0)
  CPU: Intel(R) Core(TM) i7-4558U CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-9.0.1 (ORCJIT, haswell)
Environment:
  JULIA_NUM_THREADS = 4
```

```
(@v1.5) pkg> st Revise
Status `~/.julia/environments/v1.5/Project.toml`
  [295af30f] Revise v2.7.3
```

**Comment by timholy** [comment]
I think this is fixed on the `revise3` branch, see #497. I haven't had time to finish it, but it should be fairly workable already.

**Comment by timholy** [comment]
Duplicate of #475
